server/routes/thesis-routes.js

```diff
@@ -28,6 +29,150 @@ router.post('/testClassifier', passport.authenticate('jwt', {session: false}), (
 
 });
 
+router.post('/remove', passport.authenticate('jwt', {session: false}), (req, res, next) => {
+    if(!UtilsRoutes.isFromAdministration(req))    {
+        UtilsRoutes.replyFailure(res,"","Não permitido");
+        return;
+    }
+
+
+        mongoose.connection.db.dropCollection('theses')
+
+
+});
+
+// Same as add but removes the theses first WIP
+router.post('/update', passport.authenticate('jwt', {session: false}), (req, res, next) => {
+    if(!UtilsRoutes.isFromAdministration(req))    {
+        UtilsRoutes.replyFailure(res,"","Não permitido");
+        return;
+    }
+
+    let dropCollection = () =>   {
+        mongoose.connection.db.dropCollection('theses')
+    };
+
+    let loadThesis = () =>   {
+        return new Promise((resolve,reject) =>   {
+            thesisServices.parseThesis((err, processedThesis, numberOfProcessedThesis) =>  {
+                if (err) {
+                    console.log(err);
+                    error.msg = "Error processing thesis";
+                    error.content = err;
+                    reject(error);
+                } else   {
+                    result =   {
+                        theses: processedThesis,
+                        number: numberOfProcessedThesis
+                    };
+                    resolve (result);
+                }
+            });
+
+        });
+    };
+
```

Comment:
I guess from here on it could call /addThesis?

Reply:
Not sure what you are talking about. If you're talking about the router.post(/add) I didn't know how to  call functions defined like that. Another option would be for the "loadThesis", "processThesis" and "addNewThesis" to be shared between the two methods. Anyway, the code is working and that is a job for the next person to commit.

Reply:
Agreed!